### PR TITLE
support for Play 2.8.16+

### PR DESCRIPTION
**Pull request by tbradellis**
The play 2.7 instrumentation module stopped being applied to Play on version 2.8.16.

[Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag]
Weave violations reported on the test:

WeaveViolation{type=METHOD_NEW_CALL_ORIGINAL_UNSUPPORTED, clazz=play/core/server/Server$, method=getHandlerFor(Lplay/api/mvc/RequestHeader;Lscala/util/Try;)Lscala/Tuple2;}
WeaveViolation{type=METHOD_NEW_NON_PRIVATE_UNSUPPORTED, clazz=play/core/server/Server$, method=getHandlerFor(Lplay/api/mvc/RequestHeader;Lscala/util/Try;)Lscala/Tuple2;}

github play compare versions:
Indeed, the signature has changed:
https://github.com/playframework/playframework/commit/a5d83283669dd69dbdba1407d231113aee3a02e5

### Related Github Issue
https://github.com/newrelic/newrelic-java-agent/issues/874
### Testing
unit and instrumentation test included

### Checks

[ √] Are your contributions backwards compatible with relevant frameworks and APIs?
[ ] Does your code contain any breaking changes? Please describe. 
[ √] Does your code introduce any new dependencies? Please describe.
Only compile time dependencies to support instrumentation verification

### CODE REVIEW

File: instrumentation/play-2.8.16/src/main/scala/play/core/routing/HandlerInvokerFactory_Instrumentation.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/*
+ *
+ *  * Copyright 2020 New Relic Corporation. All rights reserved.
```
Comment (jasonjkeller) [inline instrumentation/play-2.8.16/src/main/scala/play/core/routing/HandlerInvokerFactory_Instrumentation.scala:3]:
```suggestion
 *  * Copyright 2022 New Relic Corporation. All rights reserved.
```

File: instrumentation/play-2.8.16/src/test/java/com/nr/agent/instrumentation/play2816/PlayApplicationServerRule.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ *  * Copyright 2020 New Relic Corporation. All rights reserved.
```
Comment (jasonjkeller) [inline instrumentation/play-2.8.16/src/test/java/com/nr/agent/instrumentation/play2816/PlayApplicationServerRule.java:3]:
```suggestion
 *  * Copyright 2022 New Relic Corporation. All rights reserved.
```

File: instrumentation/play-2.8.16/src/test/java/com/nr/agent/instrumentation/play2816/SimpleJavaController.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/*
+ *
+ *  * Copyright 2020 New Relic Corporation. All rights reserved.
```
Comment (jasonjkeller) [inline instrumentation/play-2.8.16/src/test/java/com/nr/agent/instrumentation/play2816/SimpleJavaController.java:3]:
```suggestion
 *  * Copyright 2022 New Relic Corporation. All rights reserved.
```

**Review by jasonjkeller — CHANGES_REQUESTED** [review]
A couple of minor tweaks to make.

So am I understanding correctly that this is identical to the `play-2.7` instrumentation, the only difference being that the `Server.getHandlerFor` signature changed?

**Comment by tbradellis** [comment]
> So am I understanding correctly that this is identical to the `play-2.7` instrumentation, the only difference being that the `Server.getHandlerFor` signature changed?

That, and there were dependency graph changes for the test server that @richard-gibson pointed out.

**Review by jasonjkeller — APPROVED** [review]
Looks good but I think all of the tests that you've added that have the `Java18IncompatibleTest` annotation are also going to need the `Java19IncompatibleTest` annotation or they will fail once merged into `main`.

**Pull request merged**